pkg/controller/dataimportcron-controller.go

```diff
@@ -460,6 +484,16 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 	}); err != nil {
 		return err
 	}
+	// Watch only for DataSource deletion
+	if err := c.Watch(&source.Kind{Type: &cdiv1.DataSource{}},
```

Comment:
It is possible for delete events to get lost.  If it is critical to do something when a resource is deleted, use a finalizer

Reply:
As discussed, we won't use a finalizer for DataSource, but (re-)create it in whenever we see it's missing.

Reply:
So we are fine if the delete event gets lost, and it gets recreated on the next reconcile loop?

Reply:
Yes, we found that to be the reasonable behavior.

Reply:
The question is when will that next reconcile happen?